lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+          {
+            :stack_id => stack_id
+          }
+        )
+        http_response = terraform_runner_client['api/stack/cancel'].post(
+          payload, :content_type => 'application/json'
+        )
+        _log.info("==== Cancel Stack Response: \n #{http_response.body}")
+        Terraform::Runner::Response.parsed_response(http_response)
+      end
+
+      # Wait for TerraformRunner Stack Job to complete
+      def wait_until_completes(stack_id)
+        interval_in_secs = stack_job_interval_in_secs
+        max_time_in_secs = stack_job_max_time_in_secs
+
+        response = nil
+        Timeout.timeout(max_time_in_secs) do
+          _log.debug("Starting wait for terraform-runner/stack/#{stack_id} completes ...")
+          i = 0
+          loop do
+            _log.debug("loop #{i}")
+            i += 1
+
+            response = retrieve_stack_job(stack_id)
+
+            _log.info("status: #{response.status}")
+
+            case response.status
+            when "SUCCESS"
+              _log.debug("Successful! (stack_job/:#{stack_id})")
+              break
+
+            when "FAILED"
+              _log.info("Failed! (stack_job/:#{stack_id} fails!)")
+              _log.info(response.error_message)
+              break
+
+            when nil
+              _log.info("No status! stack_job/:#{stack_id} must have failed, check response ...")
+              _log.info(response.message)
+              break
+            end
+            _log.info("============\n stack_job/:#{stack_id} status=#{response.status} \n============")
+
+            # sleep interval
+            _log.debug("Sleep for #{interval_in_secs} secs")
+            sleep interval_in_secs
+
+            break unless i < 20
+          end
+          _log.debug("loop ends: ran #{i} times")
+        end
+        response
+      end
+
+      # Create TerraformRunner Stack Job, wait until completes
+      def create_stack_job_and_wait_until_completes(
+        template_path,
+        input_vars: [],
+        tags: nil,
+        credentials: [],
+        env_vars: {},
+        name: "stack-#{rand(36**8).to_s(36)}"
+      )
+        _log.info("create_stack_job_and_wait_until_completes for #{template_path}")
+        response = create_stack_job(
+          template_path,
+          :input_vars  => input_vars,
+          :tags        => tags,
+          :credentials => credentials,
+          :env_vars    => env_vars,
+          :name        => name
+        )
+        wait_until_completes(response.stack_id)
+      end
+
+      # create zip from directory
+      def create_zip_file_from_directory(zip_file_path, template_path)
+        dir_path = template_path # directory to be zipped
+        dir_path = path[0...-1] if dir_path.end_with?('/')
+
+        _log.debug("Create #{zip_file_path}")
+        Zip::File.open(zip_file_path, Zip::File::CREATE) do |zipfile|
+          Dir.chdir(dir_path)
+          Dir.glob("**/*").reject { |fn| File.directory?(fn) }.each do |file|
```

Comment:
I'd prefer selecting files vs. rejecting directories.

```suggestion
          Dir.glob("**/*").select { |f| File.file?(f) }.each do |file|
```

Reply:
done